### DISCUSSION TITLE
new stable version?

### DISCUSSION THREAD

**Discussion by TecnoTalarn** [Hamlib/Hamlib#1605] [General]
Hello, is it possible to find a newer version of hamlib that can properly support the new equipment?
I have an Icom ic7300 that works very well with many programs via hamlib, but I also have a Yaesu FT710 that with hamlib 4.5.5 does not work well.
The Wsjt-X software has been compiled with a 4.6 git version and works perfectly with my yaesu, but other software such as Qlog, does not quite work with the official version 4.5.5. In fact, the best they work is using the parameters of the Yaesu FTDX10, but it does not understand some things and gives errors.
It might be time to make a new, stable version with all the changes you've been making over the last year.
There are new radios every day, and mostly bug fixes that are not in the stable version.
thank you

**Comment by mdblack98** [comment]
You need to contact the QLog people -- they chose to do an implementation that is not forwards compatible.
Hamlib will not change to support what they are doing as it would cripple new features in Hamlib.

Mike W9MDB









On Sunday, September 1, 2024 at 09:21:33 AM CDT, Jordi Vidal ***@***.***> wrote: 







Hello, is it possible to find a newer version of hamlib that can properly support the new equipment?
I have an Icom ic7300 that works very well with many programs via hamlib, but I also have a Yaesu FT710 that with hamlib 4.5.5 does not work well.
The Wsjt-X software has been compiled with a 4.6 git version and works perfectly with my yaesu, but other software such as Qlog, does not quite work with the official version 4.5.5. In fact, the best they work is using the parameters of the Yaesu FTDX10, but it does not understand some things and gives errors.
It might be time to make a new, stable version with all the changes you've been making over the last year.
There are new radios every day, and mostly bug fixes that are not in the stable version.
thank you

**Reply by foldynl** [reply]
Hi Mike,

I believe a better question would be: Is it possible to release a new stable version based on the 4.6.x branch? The last official release is 4.5.5, which is over a year old.

Of course, Qlog is prepared for the 4.6.x version, but installation packages have not been built with this version because 4.6.x is still unofficial.

I think many other applications would appreciate having a predictable development cycle for such an important library as Hamlib. That's why I would like to ask if it's time to stabilize the 4.6.x branch and release it as an official version.

**Comment by mdblack98** [comment]
Not until all known bugs are fixed.Mike W9MDBSent from my iPhoneOn Sep 1, 2024, at 12:36 PM, Ladislav ***@***.***> wrote:﻿
Hi Mike,
I believe a better question would be: Is it possible to release a new stable version based on the 4.6.x branch? The last official release is 4.5.5, which is over a year old.
Of course, Qlog is prepared for the 4.6.x version, but installation packages have not been built with this version because 4.6.x is still unofficial.
I think many other applications would appreciate having a predictable development cycle for such an important library as Hamlib. That's why I would like to ask if it's time to stabilize the 4.6.x branch and release it as an official version.

**Reply by foldynl** [reply]
Please consider revisiting this decision. I'm not suggesting that the library should be released with bugs, but rather that the development cycle itself could be adjusted. Hamlib is the only viable library for controlling hamradio ardware in Linux. No hamradio software that interfaces with hardware can function without it. However, waiting to release the library until "all known bugs are fixed" is quite limiting because:

1) Linux distributions only include stable release versions. This means version 4.5.5, which is already quite outdated.
2) As I understand it, the 4.5.5 branch is no longer maintained, as all fixes are committed to the master branch. So, the bugs in 4.5.5 will never be fixed before releasing 4.6
3) The master branch represents the upcoming 4.6 version, which is still in development. This makes it unsuitable for use in applications because its features can change from one commit to the next.
4) New hardware support is only being added to the master (4.6), but there’s no clear indication of when this version will be released.This means the only way to use Hamlib with the latest fixes is to use the master branch and link it statically. However, this goes against the "good development practices" for Linux, where shared libraries should be used when available under the particular distribution.

The reasons mentioned above are not a call to release buggy versions of Hamlib. Instead, it’s a request to consider releasing patch versions more deterministically. The NEWS file is growing, the release is large, and new rigs are constantly being added. This doesn’t seem to suggest that 4.6 will be released in the near future, does it?

**Comment by TecnoTalarn** [comment]
"Not until all known bugs are fixed"
Does this have a specific date? Because there will surely be endless problems.

As with new radios, we will also have them every few months.

I don't understand this way of doing it, and it might be better to update the stable version every X months with whatever gets fixed during those months, rather than waiting for EVERYTHING to be fixed before making a new version.

How many bugs have been fixed since the last stable release? And while we can't enjoy it. The work you do is impressive, no doubt about it, but those of us with new radios can't enjoy it with stable versions. A "rollup" of fixes every X months would help a lot.


Greetings!

**Comment by N0NB** [comment]
4.6 released.
